### PR TITLE
Fix cover option to Title Section

### PR DESCRIPTION
**Pull request by MV88**
## Description
Add possibily to apply cover style to a title

## Issues
 - #4259

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
title cannot be cover

**What is the new behavior?**
title can be cover

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: build/buildConfig.js [hunk 1]
```diff
@@ -184,6 +184,13 @@ module.exports = (bundles, themeEntries, paths, extractThemesPlugin, prod, publi
                     host: "dev.mapstore.geo-solutions.it"
                 }
             },
+            '/mapstore/rest/geostore': {
```
Comment (offtherailz) [inline build/buildConfig.js]: you shouldn't commit this. /rest/geostore above is more than enough

File: web/client/components/geostory/layouts/sections/Section.jsx [hunk 2]
```diff
@@ -72,6 +74,7 @@ class Section extends React.Component {
                 update={this.props.update}
                 inViewRef={this.props.inViewRef}
                 editMedia={this.props.editMedia}
+                updateSection={(element) => this.props.updateSection(`sections[{"id": "${this.props.id}"}]`, element, "merge")}
```
Comment (offtherailz) [inline web/client/components/geostory/layouts/sections/Section.jsx]: update should be enough. It's already mapped to `sections[{id...}]` by the enhancer

File: web/client/components/geostory/layouts/sections/Title.jsx [hunk 3]
```diff
@@ -33,6 +33,7 @@ export default backgroundPropWithHandler(({
     inViewRef,
     add = () => {},
     update = () => {},
+    updateSection = () => {},
```
Comment (offtherailz) [inline web/client/components/geostory/layouts/sections/Title.jsx]: remove update section, update is enough

File: web/client/components/geostory/layouts/sections/Title.jsx [hunk 4]
```diff
@@ -59,6 +60,8 @@ export default backgroundPropWithHandler(({
                     update={updateBackground}
                     add={add}
                     editMedia={editMedia}
+                    updateSection={updateSection}
```
Comment (offtherailz) [inline web/client/components/geostory/layouts/sections/Title.jsx]: remove

File: web/client/components/geostory/layouts/sections/Title.jsx [hunk 5]
```diff
@@ -81,6 +84,7 @@ export default backgroundPropWithHandler(({
             mode={mode}
             add={add}
             sectionType={sectionType}
+            updateSection={updateSection}
```
Comment (offtherailz) [inline web/client/components/geostory/layouts/sections/Title.jsx]: remove

File: web/client/plugins/GeoStory.jsx [hunk 6]
```diff
@@ -52,6 +52,7 @@ export default createPlugin("GeoStory", {
             }), {
                 add,
                 update,
+                updateSection: update,
```
Comment (offtherailz) [inline web/client/plugins/GeoStory.jsx]: remove

File: web/client/reducers/mediaEditor.js [hunk 7]
```diff
@@ -60,7 +60,7 @@ export const DEFAULT_STATE = {
             geostoreMap: {
                 name: "Geostore Dev",
                 type: SourceTypes.GEOSTORE,
-                baseURL: "/rest/geostore/",
+                baseURL: "/mapstore/rest/geostore/",
```
Comment (offtherailz) [inline web/client/reducers/mediaEditor.js]: keep it unchanged

File: web/client/components/geostory/layouts/sections/Section.jsx [hunk 8]
```diff
@@ -35,6 +35,7 @@ class Section extends React.Component {
         storyType: PropTypes.oneOf(lists.StoryTypes),
         add: PropTypes.func,
         update: PropTypes.func,
+        updateSection: PropTypes.func,
```
Comment (offtherailz) [inline web/client/components/geostory/layouts/sections/Section.jsx]: no need, update method is already enhanced accordingly to map to the section

File: web/client/components/geostory/layouts/Cascade.jsx [hunk 9]
```diff
@@ -72,6 +72,7 @@ const Cascade = ({
     sections = [],
     add = () => {},
     onVisibilityChange = () => {},
+    updateSection = () => {},
```
Comment (offtherailz) [inline web/client/components/geostory/layouts/Cascade.jsx]: remove

File: web/client/components/geostory/layouts/Cascade.jsx [hunk 10]
```diff
@@ -91,6 +92,7 @@ const Cascade = ({
                                 add={add}
                                 editMedia={editMedia}
                                 update={update}
+                                updateSection={updateSection}
```
Comment (offtherailz) [inline web/client/components/geostory/layouts/Cascade.jsx]: remove

File: web/client/components/geostory/layouts/sections/Background.jsx [hunk 11]
```diff
@@ -37,6 +37,7 @@ class Background extends Component {
         width: PropTypes.number,
         tools: PropTypes.object,
         style: PropTypes.object,
+        updateSection: PropTypes.func,
```
Comment (offtherailz) [inline web/client/components/geostory/layouts/sections/Background.jsx]: remove

File: web/client/components/geostory/layouts/Cascade.jsx (hunk 9 above)
Comment (MV88) [reply web/client/components/geostory/layouts/Cascade.jsx]: it does not work if i remove this

File: web/client/components/geostory/layouts/sections/Section.jsx (hunk 2 above)
Comment (MV88) [reply web/client/components/geostory/layouts/sections/Section.jsx]: in this case i need an update where the path works only for sections

**Pull request merged**